### PR TITLE
Initial vue.js implementation for variant filtration (#563)

### PR DESCRIPTION
**Pull request by stolpeo**
Related-Issue: #563
Projected-result-impact: none

### CODE REVIEW

**Review by holtgrewe — CHANGES_REQUESTED** [review]
This is missing an `make isort black flake8`, I believe.

File: .gitignore [hunk 1]
```diff
@@ -355,6 +355,7 @@ mailhog
 varfish/media/
 
 src/
+!variants/vueapp/src/
```
Comment (holtgrewe) [inline .gitignore:358]: Probably, the above line should be `/src`, then your addition is not necessary.

**Review by holtgrewe — COMMENTED** [review]
Can you run the formatter `prettier` on your code (and add it to your `package.json`)?

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,386 @@
+{
```
Comment (holtgrewe) [inline package-lock.json:1]: I don't think that we want to add a global package.json/package-lock.json and you should not need to touch varfish/vueapp/*

File: varfish/vueapp/package.json [hunk 3]
```diff
@@ -6,11 +6,14 @@
     "serve": "vue-cli-service serve",
```
Comment (holtgrewe) [inline varfish/vueapp/package.json:6]: Or is this only for storybook? That's OK, I guess.

File: varfish/vueapp/webpack-stats.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"status":"error","publicPath":"http://localhost:8080/","error":"ModuleParseError","message":"Module parse failed: Identifier directly after number (287:19)\nFile was processed with these loaders:\n * ./node_modules/cache-loader/dist/cjs.js\n * ./node_modules/babel-loader/lib/index.js\n * ./node_modules/cache-loader/dist/cjs.js\n * ./node_modules/vue-loader/lib/index.js\nYou may need an additional loader to handle the result of these loaders.\n|      */\n|     modalUserAnnotations() {\n>       const c = 300000000; // longer than longest chromosome\n| \n|       const ua = this.rawModalUserAnnotations;"}
```
Comment (holtgrewe) [inline varfish/vueapp/webpack-stats.json:1]: We should probably ignore this file in git?

File: variants/vueapp/.eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "root": true,
+  "extends": [
+    "plugin:vue/vue3-essential",
+    "eslint:recommended",
+    "@vue/eslint-config-prettier"
+  ],
+  "overrides": [
+    {
+      "files": [
+        "cypress/e2e/**.{cy,spec}.{js,ts,jsx,tsx}"
```
Comment (holtgrewe) [inline variants/vueapp/.eslintrc.cjs:14]: Is this necessary? We don't use typescript, and `.vue` but not `.{j,t}sx`.

File: variants/vueapp/.vscode/extensions.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (holtgrewe) [inline variants/vueapp/.vscode/extensions.json:1]: Do we really want this? Are we not mainly using PyCharm?

File: variants/vueapp/cypress.config.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const { defineConfig } = require('cypress')
+
+module.exports = defineConfig({
+  e2e: {
+    specPattern: 'cypress/e2e/**/*.{cy,spec}.{js,jsx,ts,tsx}',
```
Comment (holtgrewe) [inline variants/vueapp/cypress.config.js:5]: Similar?

File: variants/vueapp/cypress/e2e/example.cy.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+// https://docs.cypress.io/api/introduction/api.html
```
Comment (holtgrewe) [inline variants/vueapp/cypress/e2e/example.cy.js:1]: Delete file?

File: variants/vueapp/cypress/e2e/jsconfig.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (holtgrewe) [inline variants/vueapp/cypress/e2e/jsconfig.json:1]: Do we really want cypress? Can't we do it with jest (I'm using jest in my upcoming vue tests and Mikko is using this for sodar-server, I believe). Maybe remove this for now?

File: variants/vueapp/cypress/fixtures/example.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (holtgrewe) [inline variants/vueapp/cypress/fixtures/example.json:1]: This "maybe remove" comment applies to everything in cypress, I believe.

File: variants/vueapp/package.json [hunk 11]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (holtgrewe) [inline variants/vueapp/package.json:1]: remove favicon file (cannot comment on that file)

File: variants/vueapp/src/components/SmallVariantDetails.vue [hunk 12]
```diff
@@ -0,0 +1,233 @@
+<template>
```
Comment (holtgrewe) [inline variants/vueapp/src/components/SmallVariantDetails.vue:1]: Remove logo.svg?

File: variants/vueapp/src/components/__tests__/HelloWorld.spec.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import { describe, it, expect } from 'vitest'
```
Comment (holtgrewe) [inline variants/vueapp/src/components/__tests__/HelloWorld.spec.js:1]: I think this file can go away for now.

File: variants/vueapp/src/main.js [hunk 14]
```diff
@@ -0,0 +1,1288 @@
+import { createApp } from "vue";
+import { createPinia } from "pinia";
+
+import App from "./App.vue";
+
+const app = createApp(App);
+
+// https://gist.github.com/sebleier/554280
+const stopWords = [
```
Comment (holtgrewe) [inline variants/vueapp/src/main.js:9]: Can you put this into a json file and imply import it?

**Review by holtgrewe — APPROVED** [review]
Good stuff!

Please adjust commented points (some cleanup of files, externalize some JSON), then this can be merged.

File: package-lock.json (hunk 2 above)
Comment (stolpeo) [reply package-lock.json:1]: agreed

File: varfish/vueapp/package.json (hunk 3 above)
Comment (stolpeo) [reply varfish/vueapp/package.json:6]: In the end I didn't use storybook.

File: variants/vueapp/.eslintrc.cjs (hunk 5 above)
Comment (stolpeo) [reply variants/vueapp/.eslintrc.cjs:14]: I used ESLint for formatting

File: variants/vueapp/.vscode/extensions.json (hunk 6 above)
Comment (stolpeo) [reply variants/vueapp/.vscode/extensions.json:1]: Was added by default, didn't notice

File: variants/vueapp/cypress/e2e/jsconfig.json (hunk 9 above)
Comment (stolpeo) [reply variants/vueapp/cypress/e2e/jsconfig.json:1]: OK, was added by default

File: variants/vueapp/cypress/fixtures/example.json (hunk 10 above)
Comment (stolpeo) [reply variants/vueapp/cypress/fixtures/example.json:1]: cypress removed

File: variants/vueapp/.vscode/extensions.json (hunk 6 above)
Comment (stolpeo) [reply variants/vueapp/.vscode/extensions.json:1]: removed

**Pull request merged**